### PR TITLE
DOCSP-35892: Quick Start docs

### PR DESCRIPTION
**Pull request by ccho-mongodb**
Quick Start Tutorial

JIRA: https://jira.mongodb.org/browse/DOCSP-35892

Staging:
[Quick Start](https://preview-mongodbcchomongodb.gatsbyjs.io/laravel/DOCSP-35892-quick-start-docs/quick-start/)
[Driver Landing Page Summary](https://preview-mongodbcchomongodb.gatsbyjs.io/laravel/DOCSP-35892-quick-start-docs/)

Note to reviewer:
The [download repo](https://github.com/mongodb-university/laravel-quickstart) is still in progress and will be captured in another ticket.

### Checklist

- [ ] Add tests and ensure they pass
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Update documentation for new features

### CODE REVIEW

File: docs/quick-start.txt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Overview
+--------
+
+This guide shows you how to add {+odm-long+} to a new Laravel web application,
+connect to a MongoDB cluster hosted on MongoDB Atlas, and how to perform
```
Comment (norareidy) [inline docs/quick-start.txt]: S: I don't think you need to repeat "how to" here since it's at the beginning of the list

```suggestion
connect to a MongoDB cluster hosted on MongoDB Atlas, and perform
```

File: docs/quick-start.txt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Overview
+--------
+
+This guide shows you how to add {+odm-long+} to a new Laravel web application,
```
Comment (norareidy) [inline docs/quick-start.txt]: S:

```suggestion
This guide shows you how to add the {+odm-long+} to a new Laravel web application,
```

File: docs/quick-start.txt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+   If you prefer to set up your development environment in GitHub Codespaces
+   or Docker, see the linked code repository in the
+   `How to Build a Laravel + MongoDB Back End Service <https://www.mongodb.com/developer/languages/php/laravel-mongodb-tutorial/>`__
+   MongoDB Developer Center tutorial.
+
+   You can learn how to set up a local Laravel development environment
+   and perform CRUD operations by taking the
+   :mdbu-course:`Getting Started with Laravel and MongoDB </getting-started-with-laravel-and-mongodb>`
+   MongoDB University Learning Byte.
```
Comment (norareidy) [inline docs/quick-start.txt]: S: the phrase "taking the learning byte" seems a little off to me, maybe "viewing the learning byte" or "taking the MongoDB University course" would be better (unless this is how it's normally worded)

```suggestion
   and perform CRUD operations by viewing the
   :mdbu-course:`Getting Started with Laravel and MongoDB </getting-started-with-laravel-and-mongodb>`
   MongoDB University Learning Byte.
```

File: docs/quick-start.txt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+   If you prefer to connect to MongoDB by using the PHP Library driver without
+   Laravel, see `Connecting to MongoDB <https://www.mongodb.com/docs/php-library/current/tutorial/connecting/>`__
+   in the PHP Library documentation.
+
+{+odm-short+} extends the Laravel Eloquent and Query Builder syntax to
+store and retrieve data from MongoDB.
+
+MongoDB Atlas is a fully managed cloud database service that hosts your
+MongoDB deployments. You can create your own free (no credit card
+required) MongoDB Atlas deployment by following the steps in this guide.
```
Comment (norareidy) [inline docs/quick-start.txt:48]: S: up to you, but seems a little unnecessary to mention that no credit card is required

```suggestion
MongoDB deployments. You can create your own free MongoDB Atlas deployment by following the steps in this guide.
```

File: docs/quick-start/download-and-install.txt [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+Complete the following steps to install and add the {+odm-short+} dependencies
+to a Laravel web application.
+
+.. procedure::
+   :style: connected
+
+   .. step:: Install the {+php-extension+}
+
+      The {+odm-short+} requires the {+php-extension+} to manage MongoDB
```
Comment (norareidy) [inline docs/quick-start/download-and-install.txt]: S: remove "the"

```suggestion
      {+odm-short+} requires the {+php-extension+} to manage MongoDB
```

File: docs/quick-start/download-and-install.txt [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+      connections and commands.
+      Follow the `Installing the MongoDB PHP Driver with PECL <https://www.php.net/manual/en/mongodb.installation.pecl.php>`__
+      guide to install {+php-extension+}.
```
Comment (norareidy) [inline docs/quick-start/download-and-install.txt]: s: add "the"

```suggestion
      guide to install the {+php-extension+}.
```

File: docs/quick-start/download-and-install.txt [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+   .. step:: Install Laravel
+
+      Ensure that the version of Laravel you install is compatible with the
+      version of {+odm-short+}. 
+
+
+      Then run the following command to install
+      Laravel.
```
Comment (norareidy) [inline docs/quick-start/download-and-install.txt]: S: to match the punctuation of the other instructions on this page

```suggestion
      Laravel:
```

File: docs/quick-start/download-and-install.txt [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+      .. code-block:: bash
+
+         cd {+quickstart-app-name+}
+
+      Run the following command to add the {+odm-short+} dependency to
+      your application:
+
+      .. code-block:: bash
+
+         composer require mongodb/laravel-mongodb:^{+package-version+}
```
Comment (norareidy) [inline docs/quick-start/download-and-install.txt:103]: Q: should this also say `^<version-number>` (like the code in step 2) rather than including a specific number with {`+package-version+}`?

File: docs/includes/quick-start/troubleshoot.rst [hunk 9]
```diff
@@ -0,0 +1,7 @@
+.. note::
+
+   If you run into issues on this step, ask for help in the
```
Comment (norareidy) [inline docs/includes/quick-start/troubleshoot.rst]: S: since this note is included on tutorial pages with multiple steps, I think this should say "these steps" rather than "this step"
```suggestion
   If you run into issues on these steps, ask for help in the
```

File: docs/quick-start/download-and-install.txt [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+
+      When the installation completes, you should see the
+      following line in the ``require`` object in your ``composer.json`` file:
+
+      .. code-block:: json
+         :copyable: false
+
+         "mongodb/laravel-mongodb": "^{+package-version+}"
+
+      After you complete these steps, you should have a new Laravel project
```
Comment (norareidy) [inline docs/quick-start/download-and-install.txt]: S: to match the other tutorials

```suggestion
      After you complete these steps, you have a new Laravel project
```

File: docs/quick-start/create-a-connection-string.txt [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+You can connect to your MongoDB deployment by providing a
+**connection URI**, also called a *connection string*, which
+instructs the driver on how to connect to a MongoDB deployment
+and how to behave while connected.
+
+The connection string includes the hostname or IP address and
+port of your deployment, the authentication mechanism, user credentials
+when applicable, and connection options.
+
+To connect to an instance or deployment not hosted on Atlas, see the
```
Comment (norareidy) [inline docs/quick-start/create-a-connection-string.txt]: S: remove "the"

```suggestion
To connect to an instance or deployment not hosted on Atlas, see
```

File: docs/quick-start/create-a-connection-string.txt [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+.. procedure::
+   :style: connected
+
+   .. step:: Find your MongoDB Atlas Connection String
+
+      To retrieve your connection string for the deployment that
+      you created in the :ref:`previous step <laravel-quick-start-create-deployment>`,
+      log in to your Atlas account and navigate to the
+      :guilabel:`Database` section and click the :guilabel:`Connect` button
+      for your new deployment.
```
Comment (norareidy) [inline docs/quick-start/create-a-connection-string.txt]: S: I think it would be helpful to break this sentence up

```suggestion
      log in to your Atlas account. Then, navigate to the
      :guilabel:`Database` section and click the :guilabel:`Connect` button
      for your new deployment.
```

File: docs/quick-start/create-a-connection-string.txt [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+         :alt: The connect button in the clusters section of the Atlas UI
+
+      Proceed to the :guilabel:`Connect your application` section and select
+      "PHP" from the :guilabel:`Driver` selection menu and the version
+      that best matches the version you installed from the :guilabel:`Version`
+      selection menu.
+
+      Select the :guilabel:`Password (SCRAM)` authentication mechanism.
+
+      Deselect the :guilabel:`Include full driver code example` to view
```
Comment (norareidy) [inline docs/quick-start/create-a-connection-string.txt]: S:
```suggestion
      Deselect the :guilabel:`Include full driver code example` option to view
```

File: docs/quick-start/configure-mongodb.txt [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+      in the following line:
+
+      .. code-block:: php
+
+         'default' => env('DB_CONNECTION', 'mongodb'),
+
+      Add the following highlighted ``mongodb`` entry to the ``connections`` array
+      in the same file. Replace the ``<connection string>`` placeholder with the
+      connection string that you copied from the :ref:`laravel-quick-start-connection-string`
+      step in the following code example:
```
Comment (norareidy) [inline docs/quick-start/configure-mongodb.txt:30]: S: not necessary, but could make sense to introduce the code example right after the first sentence ("Add the following highlighted..."), then move the sentence about replacing the placeholders after the snippet.

```suggestion
      in the same file:
      
      // code snippet
      
      Replace the ``<connection string>`` placeholder with the
      connection string that you copied from the :ref:`laravel-quick-start-connection-string`
      step.
```

File: docs/quick-start/configure-mongodb.txt [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+             'driver' => 'mongodb',
+             'dsn' => env('DB_URI', '<connection string>'),
+             'database' => 'sample_mflix',
+           ],
+
+         // ...
+
+      .. step:: Add the Laravel MongoDB Provider
+
+         Open the ``app.php`` in the ``config`` directory and
```
Comment (norareidy) [inline docs/quick-start/configure-mongodb.txt]: S:
```suggestion
         Open the ``app.php`` file in the ``config`` directory and
```

File: docs/quick-start/configure-mongodb.txt [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+         'connections' => [
+           'mongodb' => [
+             'driver' => 'mongodb',
+             'dsn' => env('DB_URI', '<connection string>'),
+             'database' => 'sample_mflix',
+           ],
+
+         // ...
+
+      .. step:: Add the Laravel MongoDB Provider
```
Comment (norareidy) [inline docs/quick-start/configure-mongodb.txt]: I: I think there might be an indentation issue with this step, since I don't see it on the staging page

File: docs/quick-start/view-data.txt [hunk 17]
```diff
@@ -0,0 +1,172 @@
...
+
+.. procedure::
+   :style: connected
+
+   .. step:: Create a Sample Model and Controller
+
+      Create a model called ``Movie`` to represent data from the sample
+      ``movies`` collection in your MongoDB database. You can create the
+      model with a corresponding resource controller by running the
+      following command:
```
Comment (norareidy) [inline docs/quick-start/view-data.txt]: S: since this is a tutorial, I think it makes sense to replace "you can" with something more action oriented

```suggestion
      ``movies`` collection in your MongoDB database. Run the following command
      to create the model with a corresponding resource controller:
```

File: docs/quick-start/write-data.txt [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+             $data = $request->all();
+             $movie = new Movie();
+             $movie->fill($data);
+             $movie->save();
+         }
+
+   .. step:: Add a Route for the Controller Function
+
+      Add the API route to map to the ``store()`` method in the
+      ``routes/api.php`` file.
```
Comment (norareidy) [inline docs/quick-start/write-data.txt]: S:

```suggestion
      Add the following API route to map to the ``store()`` method in the
      ``routes/api.php`` file:
```

File: docs/quick-start/write-data.txt [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+           'store'
+         ]);
+
+
+   .. step:: Update the Movie Model
+
+      Add the movie detail fields to the ``$fillable`` array
+      of the ``Movie`` model. After adding them, the ``Movie``
+      class at ``app/Models/Movie.php`` should contain the
+      following code:
```
Comment (norareidy) [inline docs/quick-start/write-data.txt]: S: I think the wording of this step can be more action-oriented:

```suggestion
      Navigate to the ``Movie`` class located at ``app/Models/Movie.php``. Then, add
      the following movie detail fields to the ``$fillable`` array of the ``Movie`` model:
```

File: docs/quick-start/write-data.txt [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+
+         class Movie extends Model
+         {
+            protected $connection = 'mongodb';
+            protected $fillable = ['title', 'year', 'runtime', 'imdb', 'plot'];
+         }
+
+   .. step:: Post a Request to the API
+
+      Create a file called ``movie.json`` and insert the following data:
```
Comment (norareidy) [inline docs/quick-start/write-data.txt:66]: Q: should this code example include how to insert the data, rather than just the data itself, or are you assuming the user already knows how to insert?

File: docs/quick-start/next-steps.txt [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+Congratulations on completing the quick start tutorial!
+
+After you complete these steps, you have a Laravel web application that
+uses the {+odm-long+} to connect to your MongoDB deployment, run a query on
+the sample data, and render a retrieved result.
+
+In this tutorial, you created a Laravel web application that uses the
+{+odm-long+} to connect to your MongoDB deployment, render the result
+of a query, and write data to a MongoDB deployment hosted on
+MongoDB Atlas.
```
Comment (norareidy) [inline docs/quick-start/next-steps.txt]: I: these two paragraphs pretty much repeat the same information, so I don't think you need both

File: docs/includes/quick-start/troubleshoot.rst (hunk 9 above)
Comment (norareidy) [reply docs/includes/quick-start/troubleshoot.rst]: unless each page of the quick start is a step, and the steps on each individual page are sub-steps?

**Review by norareidy — CHANGES_REQUESTED** [review]
Looks good, just some wording suggestions / questions!

File: docs/quick-start.txt (hunk 4 above)
Comment (ccho-mongodb) [reply docs/quick-start.txt:48]: Thanks for mentioning this. The specific wording was provided by Product a few years ago. Since we haven't received any guidance to change it, I will leave it for now.

File: docs/quick-start/download-and-install.txt (hunk 7 above)
Comment (ccho-mongodb) [reply docs/quick-start/download-and-install.txt]: Thanks for spotting this -- looks like I didn't fill out all the info I intended to include.

File: docs/quick-start/download-and-install.txt (hunk 8 above)
Comment (ccho-mongodb) [reply docs/quick-start/download-and-install.txt:103]: Thanks for raising this question. No, the other placeholder is for Laravel (composer decides which is the latest version) and this one is for the Laravel MongoDB package (tied to the version of the docs). I didn't want to confuse anyone if the version of Laravel that composer chose didn't match.

File: docs/includes/quick-start/troubleshoot.rst (hunk 9 above)
Comment (ccho-mongodb) [reply docs/includes/quick-start/troubleshoot.rst]: Good point. I'll omit "on this step" for this statement across all the pages to eliminate any confusion.

File: docs/quick-start/download-and-install.txt (hunk 10 above)
Comment (ccho-mongodb) [reply docs/quick-start/download-and-install.txt]: To me, omitting "should" makes it sound definitive conclusion, but that might not match the reader's observation. I'll try to find a different wording that avoids this.

File: docs/quick-start/configure-mongodb.txt (hunk 16 above)
Comment (ccho-mongodb) [reply docs/quick-start/configure-mongodb.txt]: Good catch! Strange that this didn't report an issue in the autobuilder log.

File: docs/quick-start/view-data.txt (hunk 17 above)
Comment (ccho-mongodb) [reply docs/quick-start/view-data.txt]: Thanks for raising this. There's some nuance in that this is one way of accomplishing the task mentioned in the previous sentence. I'll update the wording to avoid offering the choice.

File: docs/quick-start/write-data.txt (hunk 19 above)
Comment (ccho-mongodb) [reply docs/quick-start/write-data.txt]: Thanks for the suggestion. I think the current wording is action-oriented, but might be a little confusing since it states what the code should look like after updating it. This is a pattern used in other steps of the Quick Start, but I'll reword this to possibly make it easier to read.

File: docs/quick-start/write-data.txt (hunk 20 above)
Comment (ccho-mongodb) [reply docs/quick-start/write-data.txt:66]: The insert is defined in "Create the Function to Store Data". I will change the title to "Create the Function to Insert Data" to make it more apparent for anyone that might be specifically looking for that word.
This step is chronologically after that one and describes how to perform a POST request with sample data so that the application calls that insert code.

File: docs/quick-start/next-steps.txt (hunk 21 above)
Comment (ccho-mongodb) [reply docs/quick-start/next-steps.txt]: Good catch, I forgot to delete one!

File: docs/quick-start/configure-mongodb.txt (hunk 14 above)
Comment (ccho-mongodb) [reply docs/quick-start/configure-mongodb.txt:30]: I think both ways can work and there are examples of both in our documentation. I'll keep this as-is for now; open to changing it if there are any strong opinions.

File: docs/quick-start/configure-mongodb.txt [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+   :name: genre
+   :values: tutorial
+
+.. meta::
+   :keywords: test connection, runnable, code example
+
+.. procedure::
+   :style: connected
+
+   .. step:: Set the Connection String in the Database Configuration
```
Comment (norareidy) [inline docs/quick-start/configure-mongodb.txt]: I: per the [style guide](https://www.mongodb.com/docs/meta/style-guide/style/titles-and-headings/#capitalization), I think these steps should use sentence-style capitalization

Comment (norareidy) [reply docs/quick-start/configure-mongodb.txt]: Applies to the other steps/files too

File: docs/quick-start/create-a-connection-string.txt [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+      :guilabel:`Database` section and click the :guilabel:`Connect` button
+      for your new deployment.
+
+      .. figure:: /includes/figures/atlas_connection_select_cluster.png
+         :alt: The connect button in the clusters section of the Atlas UI
+
+      Proceed to the :guilabel:`Connect your application` section and select
+      "PHP" from the :guilabel:`Driver` selection menu and the version
+      that best matches the version you installed from the :guilabel:`Version`
+      selection menu.
```
Comment (norareidy) [inline docs/quick-start/create-a-connection-string.txt]: S: this sentence is a little long, think it can be broken up
```suggestion
      Proceed to the :guilabel:`Connect your application` section. Select
      "PHP" from the :guilabel:`Driver` selection menu and the version
      that best matches the version you installed from the :guilabel:`Version`
      selection menu.
```

File: docs/quick-start/create-a-deployment.txt [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+
+After completing these steps, you have a new free tier MongoDB deployment on
+Atlas, database user credentials, and sample data loaded into your database.
+
+.. note::
+
+   If you run into issues on this step, ask for help in the
+   :community-forum:`MongoDB Community Forums <>`
+   or submit feedback by using the :guilabel:`Share Feedback`
+   tab on the right or bottom right side of this page.
```
Comment (norareidy) [inline docs/quick-start/create-a-deployment.txt]: S: replace this note with the troubleshoot.rst file

```suggestion
.. include:: /includes/quick-start/troubleshoot.rst
```

File: docs/quick-start/write-data.txt [hunk 25]
```diff
@@ -0,0 +1,97 @@
...
+   :keywords: test connection, runnable, code example
+
+.. procedure::
+   :style: connected
+
+   .. step:: Create the Function to Insert Data
+
+      Update the ``store()`` method ``MovieController.php``, located in
+      the ``app/Http/Controllers`` directory to insert request data
+      into the database as shown in the following code:
```
Comment (norareidy) [inline docs/quick-start/write-data.txt]: S: I think the current wording is slighly unclear, something like this might work:
```suggestion
      Navigate to the ``MovieController.php`` file located in
      the ``app/Http/Controllers`` directory. Then, update the ``store()`` method to insert request data
      into the database as shown in the following code:
```

File: docs/quick-start/write-data.txt [hunk 26]
```diff
@@ -0,0 +1,97 @@
...
+      Send the JSON payload to the endpoint as a ``POST`` request by running
+      the following command in your shell:
+
+      .. code-block:: bash
+
+         curl -H "Content-Type: application/json" --data @movie.json http://localhost:8000/api/movies
+
+   .. step:: View the Data
+
+      Open ``http://127.0.0.1:8000/browse_movies`` in your web browser to view
```
Comment (norareidy) [inline docs/quick-start/write-data.txt]: S: nit, but since this is a URL and it's clickable in view-data.txt, could be helpful to also make this link clickable:

```suggestion
      Open http://127.0.0.1:8000/browse_movies in your web browser to view
```

File: docs/quick-start/download-and-install.txt [hunk 27]
```diff
@@ -0,0 +1,119 @@
...
+         :copyable: false
+
+         INFO  Application ready in [{+quickstart-app-name+}]. You can start your local development using:
+
+         ➜ cd {+quickstart-app-name+}
+         ➜ php artisan serve
+
+         New to Laravel? Check out our bootcamp and documentation. Build something amazing!
+
+   .. step:: Add a Laravel Application Encryption Key
```
Comment (norareidy) [inline docs/quick-start/download-and-install.txt]: Q: do you need to run this command in the `my-app` directory? If so, I'd mention that in the step

**Review by norareidy — APPROVED** [review]
LGTM with some final suggestions!

File: docs/quick-start/write-data.txt (hunk 25 above)
Comment (ccho-mongodb) [reply docs/quick-start/write-data.txt]: I can see how "navigate to" might be a step in an IDE in which you need to expand folder contents, but on the command line, there's no need to change directory to open a file for editing, so I chose to stay away from that wording.

Which part was unclear, and do you think this could be a good alternative (omits describing what they're accomplishing and focuses only on the actions they need to take to proceed):

"Replace the store() method in the MovieController.php file, located in the app/Http/Controllers" directory with the following code:"

File: docs/quick-start/download-and-install.txt (hunk 27 above)
Comment (ccho-mongodb) [reply docs/quick-start/download-and-install.txt]: Good callout! I will move the "cd" command into this step (currently in the next step) to make it clear.

File: docs/quick-start/write-data.txt (hunk 25 above)
Comment (norareidy) [reply docs/quick-start/write-data.txt]: Okay that makes sense. The main issue I had was the "update the `store()` method `MovieController.php`" part of the sentence since it seemed incomplete (but rewording to "Replace the store() method in the MovieController.php file" fixes that issue). I also think there's a missing comma after "directory,"  but otherwise the alternative wording you gave works for me!

File: .gitignore [hunk 28]
```diff
@@ -9,3 +9,4 @@ composer.phar
 phpunit.xml
 phpstan.neon
 /.cache/
+docs/.vscode
```
Comment (GromNaN) [inline .gitignore]: This looks like a configuration that should be [global to your system](https://www.mashupgarage.com/playbook/git/gitignore.html) instead of specific to every project you work on.

File: docs/quick-start/download-and-install.txt [hunk 29]
```diff
@@ -0,0 +1,119 @@
...
+
+         composer require mongodb/laravel-mongodb:^{+package-version+}
+
+      When the installation completes, verify that the ``composer.json`` file
+      includes the following line in the ``require`` object:
+
+      .. code-block:: json
+         :copyable: false
+
+         "mongodb/laravel-mongodb": "^{+package-version+}"
```
Comment (GromNaN) [inline docs/quick-start/download-and-install.txt:111]: It's good that you put the caret prefix to allow new minor versions.

**Review by GromNaN — APPROVED** [review]

File: .gitignore (hunk 28 above)
Comment (ccho-mongodb) [reply .gitignore]: Thanks for the link! I'll remove this line from the .gitignore.
